src/pudl/settings.py

```diff
@@ -139,12 +139,16 @@ class Ferc714Settings(GenericDatasetSettings):
     """
 
     data_source: ClassVar[DataSource] = DataSource.from_id("ferc714")
-
-    # Note: Only older data is currently supported. Starting in 2021 FERC-714 is being
-    # published as XBRL, and we haven't integrated it. The older data is published as
-    # monolithic CSV files, so asking for any year processes all of them.
     years: list[int] = data_source.working_partitions["years"]
 
+    # The older 714 data is distributed as CSV files and has a different extraction
+    # process than the FERC DBF extraction process.
+
+    @property
+    def xbrl_years(self):
+        """Return validated years for which XBRL data is available."""
+        return [year for year in self.years if year <= 2021]
```

Comment:
I think you mean `year >= 2021` rather than `year <= 2021` here, and it might be why the unit tests are failing.

```suggestion
        return [year for year in self.years if year >= 2021]
```

Reply:
good catch!

Reply:
Sadly not why the unit tests are failing though.

Reply:
Unit test failures have something to do with this snipit I changed in `io_managers.py`: 

```
        ferc_settings = context.resources.dataset_settings.get_datasets()[
            self.db_name.replace("_xbrl", "")
        ]
```

The `pytest test/unit/io_managers_test.py::test_ferc_xbrl_sqlite_io_manager_dedupes` function does not like this.

Reply:
Ok I fixed it! changed that snipit to: 

```
        ferc_settings = getattr(
            context.resources.dataset_settings, self.db_name.replace("_xbrl", "")
        )
```